### PR TITLE
303 org affiliation profiles

### LINKED ISSUE #303: Add "organisational affiliation" to profiles

### PR DESCRIPTION
**Pull request by sebgrebe**
closes #303 

Notice that as in [this PR](https://github.com/rtcharity/eahub.org/pull/346), django-crispy forms aren't fully implemented. Once we figure out how they can be applied to multiselect fields, we should change the form here, too. However, I'd argue that #303 can be considered closed for the purposes of the publicised launch milestone.

### CODE REVIEW

File: eahub/profiles/models.py [hunk 1]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
         ONE_FOR_THE_WORLD: "One for the World",
     }
 
+class OrganisationalAffiliations(enum.Enum):
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Class names are singular.

File: eahub/profiles/models.py [hunk 2]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        CENTER_FOR_HUMAN_COMPATIBLE_AI: "Center for Human Compatible AI",
+        CENTER_FOR_EFFECTIVE_ALTRUISM: "Center for Effective Altruism",
+        CENTER_FOR_THE_FUTURE_OF_INTELLIGENCE: "Center for the Future of Intelligence",
+        CENTER_FOR_APPLIED_RATIONALITY: "Center for Applied Rationality",
+        CENTER_FOR_THE_STUDY_OF_EXISTENTIAL_RISK: "Center for the Study of Existential Risk",
+        CHARITY_SCIENCE_HEALTH: "Charity Science Health",
+        CHARITY_ENTREPRENEURSHIP: "Charity Entrepreneurship",
+        FORETHOUGHT_FOUNDATION: "Forethought Foundation",
+        FOUNDATIONAL_RESEARCH_INSTITUTE: "Foundational Research Institute",
+        FOUNDERS_PLEDGE: "Founder's Pledge",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Founders

File: eahub/profiles/models.py [hunk 3]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+    RETHINK_PRIORITIES = 29
+    SCHOOLS_FOR_HIGH_IMPACT_CHARITY = 30
+    STIFTUNG_FUR_EFFEKTIVEN_ALTRUISMUS = 31
+
+    labels = {
+        EIGHTYTHOUSAND_HOURS: "80,000 Hours",
+        ANIMAL_CHARITY_EVALUATORS: "Animal Charity Evaluators",
+        BERKELEY_EXISTENTIAL_RISK_INITIATIVE: "Berkeley Existential Risk Initiative",
+        CENTER_FOR_HUMAN_COMPATIBLE_AI: "Center for Human Compatible AI",
+        CENTER_FOR_EFFECTIVE_ALTRUISM: "Center for Effective Altruism",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Centre (they're British)

File: eahub/profiles/models.py [hunk 4]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        CENTER_FOR_THE_FUTURE_OF_INTELLIGENCE: "Center for the Future of Intelligence",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Leverhulme Centre for the Future of Intelligence (unless this is some other org I haven't heard of)

File: eahub/profiles/models.py [hunk 5]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        CENTER_FOR_APPLIED_RATIONALITY: "Center for Applied Rationality",
+        CENTER_FOR_THE_STUDY_OF_EXISTENTIAL_RISK: "Center for the Study of Existential Risk",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Centre

File: eahub/profiles/models.py [hunk 6]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
         ONE_FOR_THE_WORLD: "One for the World",
     }
 
+class OrganisationalAffiliations(enum.Enum):
+
+    EIGHTYTHOUSAND_HOURS = 1
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Nit: `EIGHTY_THOUSAND` would be closer to standard English than `EIGHTYTHOUSAND`.

File: eahub/profiles/models.py [hunk 7]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        CHARITY_SCIENCE_HEALTH: "Charity Science Health",
+        CHARITY_ENTREPRENEURSHIP: "Charity Entrepreneurship",
+        FORETHOUGHT_FOUNDATION: "Forethought Foundation",
+        FOUNDATIONAL_RESEARCH_INSTITUTE: "Foundational Research Institute",
+        FOUNDERS_PLEDGE: "Founder's Pledge",
+        FUTURE_OF_LIFE_INSTITUTE: "Future of Life Institute",
+        GIVEWELL: "GiveWell",
+        GLOBAL_CATASTROPHIC_RISK_INSTITUTE: "Global Catastrophic Risk Institute",
+        GLOBAL_PRIORITIES_INSTITUTE: "Global Priorities Institute",
+        GOOD_FOOD_INSTITUTE: "Good Food Institute",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: The Good Food Institute

File: eahub/profiles/models.py [hunk 8]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        LOCAL_EFFECTIVE_ALTRUISM_NETWORK: "Local Effective Altruism Network",
+        MACHINE_INTELLIGENCE_RESEARCH_INSTITUTE: "Machine Intelligence Research Institute",
+        ONE_FOR_THE_WORLD: "One For The World",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: One for the World

File: eahub/profiles/models.py [hunk 9]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        OPEN_PHILANTHROPY_PROJECT: "Open Philanthropy Project",
+        RAISING_FOR_EFFECTIVE_GIVING: "Raising for Effective Giving",
+        RETHINK_CHARITY: "Rethink Charity",
+        RETHINK_CHARITY_FORWARD: "Rethink Charity Forward",
+        RETHINK_PRIORITIES: "Rethink Priorities",
+        SCHOOLS_FOR_HIGH_IMPACT_CHARITY: "Schools for High Impact Charity",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: Students for High-Impact Charity

File: eahub/profiles/models.py [hunk 10]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+        FUTURE_OF_LIFE_INSTITUTE: "Future of Life Institute",
+        GIVEWELL: "GiveWell",
+        GLOBAL_CATASTROPHIC_RISK_INSTITUTE: "Global Catastrophic Risk Institute",
+        GLOBAL_PRIORITIES_INSTITUTE: "Global Priorities Institute",
+        GOOD_FOOD_INSTITUTE: "Good Food Institute",
+        LOCAL_EFFECTIVE_ALTRUISM_NETWORK: "Local Effective Altruism Network",
+        MACHINE_INTELLIGENCE_RESEARCH_INSTITUTE: "Machine Intelligence Research Institute",
+        ONE_FOR_THE_WORLD: "One For The World",
+        OPEN_PHILANTHROPY_PROJECT: "Open Philanthropy Project",
+        RAISING_FOR_EFFECTIVE_GIVING: "Raising for Effective Giving",
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: This doesn't have its own staff separate from EAF. What were the criteria for something to be on here?

File: eahub/profiles/models.py [hunk 11]
```diff
@@ -68,6 +68,75 @@ class GivingPledge(enum.Enum):
...
+    EIGHTYTHOUSAND_HOURS = 1
+    ANIMAL_CHARITY_EVALUATORS = 2
+    BERKELEY_EXISTENTIAL_RISK_INITIATIVE = 3
+    CENTER_FOR_HUMAN_COMPATIBLE_AI = 4
+    CENTER_FOR_EFFECTIVE_ALTRUISM = 5
+    CENTER_FOR_THE_FUTURE_OF_INTELLIGENCE = 6
+    CENTER_FOR_APPLIED_RATIONALITY = 7
+    CENTER_FOR_THE_STUDY_OF_EXISTENTIAL_RISK = 8
+    CHARITY_SCIENCE_HEALTH = 9
+    CHARITY_ENTREPRENEURSHIP = 10
```
Comment (taymonbeal) [inline eahub/profiles/models.py]: This looks out of alphabetical order.

File: eahub/profiles/migrations/0001_initial.py [hunk 12]
```diff
@@ -80,6 +80,17 @@ class Migration(migrations.Migration):
                     "available_as_speaker",
                     models.BooleanField(blank=True, default=None, null=True),
                 ),
+                (
```
Comment (taymonbeal) [inline eahub/profiles/migrations/0001_initial.py]: Unless we're going to do another data wipe, we can't change old migrations, we have to create and run a new one. This is what the `makemigrations` command does.

File: eahub/profiles/models.py (hunk 1 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: changed

File: eahub/profiles/models.py (hunk 2 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: changed

File: eahub/profiles/models.py (hunk 4 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: changed

File: eahub/profiles/models.py (hunk 8 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: Changed

File: eahub/profiles/models.py (hunk 10 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: @richenda?

File: eahub/profiles/models.py (hunk 11 above)
Comment (sebgrebe) [reply eahub/profiles/models.py]: Changed

File: eahub/profiles/models.py (hunk 1 above)
Comment (taymonbeal) [reply eahub/profiles/models.py]: It doesn't look changed to me.

Comment (sebgrebe) [reply eahub/profiles/models.py]: haven't pushed yet, will in a minute

File: eahub/profiles/models.py (hunk 10 above)
Comment (taymonbeal) [reply eahub/profiles/models.py]: Apparently this is supposed to be left as is.

Comment (Richenda) [reply eahub/profiles/models.py]: This is the list that is used in the EA Newsletter and which is CEA approved, as Seb says. Even if it's one org too many or unecessary duplication, that kind of fine grain really doesn't matter right now. Getting this up and ready fast is the most important thing. Particularly as we are definitely dropping this drop down menu entirely as soon as we have built the EA Projects feature, which will work differently and replace this altogether. This is a stopgap.

**Pull request merged**